rules/next.js

```diff
@@ -5,5 +5,13 @@ module.exports = {
     // https://github.com/vercel/next.js/blob/v14.1.3/packages/eslint-plugin-next/src/index.ts#L56
     'plugin:@next/next/core-web-vitals',
   ],
-  rules: {},
+
+  overrides: [
+    {
+      files: ['**/pages/**/*.@(tsx|jsx)', '*.page.@(tsx|jsx)'],
```

Comment:
Any components for pages or layouts within `app/` also must be default export.

```suggestion
      files: ['**/@(app|pages)/**/*.@(tsx|jsx)', '*.page.@(tsx|jsx)'],
```

Reply:
fix: [c0dffeb](https://github.com/moneyforward/eslint-config-moneyforward/pull/243/commits/c0dffebdf1f543b4861b3d886cf71a945f8b8250)

To be precise, the target files are named `page` or `layout`, so specify these.
https://nextjs.org/docs/app/building-your-application/routing/pages-and-layouts

Reply:
@wakamsha I'm not sure but it looks like there are some files should be used default export other than `page` or `layout`, such as `not-found` and `template`, etc. (ref: https://nextjs.org/docs/app/api-reference/file-conventions)

The config should add them to the target that disables enforcing default export, or just add the entire `app` directory. It's too detailed and low-maintainable to specify all file names explicitly, I think it's better to add `app` to disable it.

Reply:
fix: [b1e20b4](https://github.com/moneyforward/eslint-config-moneyforward/pull/243/commits/b1e20b4032d0b8c62474b2d201b9c3303fe6a3e0)

@taigakiyokawa 
At this level, it's not very detailed.
The specifications of Next.js are clearly stated, so maintainability is not compromised.

Reply:
@wakamsha No, it's too detailed. It's too deep into Next.js' file conventions despite it's just an ESLint config. The conventions are clearly at now, but they have a possibility of changing and adding new files. Are you going to update this target every time a new file is added or current conventions are changed? 

Disabling the rule to all files in `app` is much more simple. After all, most files in `app` are named following the conventions so they should use default export. There is no need to specify the detailed file names following the conventions.

Reply:
@taigakiyokawa 
I don't intend to disable this rule for all files.

This rule applies to internal reference utility functions and internal components placed in the `app` directory. That's why I set it up in detail.

Reply:
@wakamsha By the way, there are also still needed to add some more files: [`global-error`](https://nextjs.org/docs/app/api-reference/file-conventions/error#global-errorjs), [`robots`](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/robots#generate-a-robots-file), [`sitemap`](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#generating-a-sitemap-using-code-js-ts), and [`icon`](https://nextjs.org/docs/app/api-reference/file-conventions/metadata/app-icons#generate-icons-using-code-js-ts-tsx).

As evidenced by that we haven't been perfectly sure which files need to use default export, specifying all files explicitly is not practical. Even if we added these files above, we can't say there are no more target files other than those we've set.

I fully understand that you really want our users to enforce this rule, but it seems this doesn't match the design of Next.js. We should compromise to disable it in the entire `app` directory. Please keep it practical and simple, and do not bother users.

Reply:
fix: [0eb4b1f](https://github.com/moneyforward/eslint-config-moneyforward/pull/243/commits/0eb4b1fd1176f2e0cde739c8241be202a5d343ad)